builtin/array.mbt

```diff
@@ -577,6 +577,66 @@ pub fn search[T : Eq](self : Array[T], value : T) -> Int? {
   }
 }
 
+/// Binary search the array index for a given element in a sorted array.
```

Comment:
Same wording issue here

```
/// Performs a binary search on a sorted array to find the index of a given element.
/// 
/// If the element is found, an `Ok` variant is returned, containing the index of the matching element in the array. 
/// If there are multiple matches, the leftmost match will be returned. 
/// 
/// If the element is not found, an `Err` variant is returned, containing the index where the element could be inserted to maintain the sorted order.
/// 
/// The array must be sorted in increasing order. If the array is not sorted, the result of this function is undefined and should not be relied upon.
/// 
/// # Example
/// ```
/// let v = [3, 4, 5];
/// let result = v.binary_search(&3);
/// assert_eq!(result, Ok(0)); // The element 3 is found at index 0
/// ```
/// 
/// # Arguments
/// - `self`: The sorted array in which to perform the search.
/// - `target`: The element to search for in the array.
/// 
/// # Returns
/// - `Result<usize, usize>`: If the element is found, `Ok` is returned with the index of the element.
/// If the element is not found, `Err` is returned with the index where the element could be inserted to maintain sorted order.
/// 
/// # Notes
/// - Ensure that the array is sorted in increasing order before calling this function.
/// - The function performs a logarithmic search, making it efficient for large arrays.
```